docs/eventModelling.md

```diff
@@ -0,0 +1,48 @@
+A user downloads an executable program from a website (reported by web proxy).|```<View>``` of type ```<Resource>``` with ```<MimeType>``` assigned appropriately.|It is not possible for the web proxy to accurately infer what a client did with the accessed resource.  Of course, analytics are likely to use the MIME type information in conjunction with other events to identify potential issues. 
+User defines a new product line within the stock control system.|```<Create>``` of type ```<Object>``` with ```<Type>``` element assigned to an appropriate string literal, e.g. "Product Line"|
+User opens a file from a network share using MSWord.exe.| ```<View>``` of type ```<File>```|*A future update to the schema will allow ```<EventSource>``` to record details of the opening application.  Currently, this must be recorded within ```<Data>``` elements of ```<EventSource>```*
+User deletes a file from a network share using the UNIX command line.|```<Delete>``` of type ```<File>```|
+DHCP service leases an IP address to a device with a certain MAC address.|```<NetworK>``` of type ```<Bind>```.  Subelements of ```<Bind>``` define the MAC address of the device within ```<Source>``` and the assigned IP address within ```<Destination>```|
+User initiates an antivirus scan.|```<AntiMalware>``` of type ```<StartScan>```. | *This will be modelled as ```<Process>``` in a future version of the schema.*
+RADIUS server allows an 802.1x authenticated device onto the network.|```<Authenticate>``` with the ```<Client>``` subelement of ```<EventSource>``` assigned to the authenticated device.|
+User searches for documents containing a particular term using enterprise search tool.| ```<Search>``` with the term specified within ```<Query>``` subelement, one of ```<Simple>```, ```<Advanced>``` or ```<Raw>```|
+User creates a contact in a messaging application.|```<Create>``` of type ```<Object>```  with ```<Type>``` element assigned to an appropriate string literal, e.g. "Contact"|
+User creates a status message in a social media application.|```<Create>``` of type ```<GroupChat>```  with ```<Type>``` element assigned to an appropriate string literal, e.g. "Status"|This assumes that other users can comment of the status message and it is therefore a converational medium, and illustrates how analogy can be used to find the most appropriate way to model events.
+User likes another user's status message in social media application.|```<Create>``` of type ```<Resource>```  with ```<Type>``` element assigned to an appropriate string literal, e.g. "Like".  ```<URL>``` can be used to indicate the actual status message and ```<User>``` subelement of ```<Permissions>``` can reflect the user who created the status update message.| *A future version of the schema will allow all Object types to refer to other instances of Object type.  In this way the actual Status message can be included within this event, rather than only indirectly by URL.*
```

Comment:
Could model this an an Association where the ossociation is of type 'Like' and the objects being associated are the User (i.e. their account in the social media app) and the Resource (the post being liked)